Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -69,46 +170,85 @@ class CDMPrivateOpenCDM : public CDMPrivate {
 
 private:
     String m_openCDMKeySystem;
-    media::OpenCdm m_openCDM;
+    // This owns OCDM Accessor and passes a bare pointer further because it's owned by CDMInstance
+    // which lives as long as any MediaKeySession lives.
+    ScopedOCDMAccessor m_openCDMAccessor;
 };
 
-class CDMInstanceOpenCDM::Session : public ThreadSafeRefCounted<CDMInstanceOpenCDM::Session> {
+class CDMInstanceOpenCDM::Session : public ThreadSafeRefCounted<CDMInstanceOpenCDM::Session>, public OCDMSessionCallbacksNotifier::Observer {
 public:
-    static Ref<Session> create(const media::OpenCdm& source, Ref<WebCore::SharedBuffer>&& initData);
+    using ChallengeGeneratedCallback = Function<void(Session*)>;
+    using KeyUpdatedCallback = Function<void(Session*, OCDMKeyStatus, const String&)>;
```

Comment:
As I mentioned above, is the `String` parameter needed or can we peel it off the onion?

Reply:
See above :)